### PR TITLE
feat: added  default message for empty group fields

### LINKED ISSUE #3380: feat: add a default message for empty group fields
**Is your feature request related to a problem? Please describe.**
For empty group members, mentors and assignments, the group page needs to have some default message:
- Mentors: `There are no mentors in the group, click "+ Add Mentors" button to add mentors.`
- Members: `There are no members in the group, click "+ Add Member" button to add members.`
- Assignments: `There are no assignments in the group, click "+ Add Assignments" button to start an assignment.`

Screenshot from 2022-12-07 22-46-51

**Additional context**
The assignment section should not show the table for empty data.

**Are you working on this? (Yes/No)**
No

### PR DESCRIPTION
**Pull request by VaibhavUpreti**
Fixes #3380 

#### Describe the changes you have made in this PR -
- Added conditions for displaying a default message for empty group field

### Screenshots of the changes (If any) -
f

Note: Please check **Allow edits from maintainers.** if you would like us to assist in the PR.

### CODE REVIEW

File: app/assets/stylesheets/groups.scss [hunk 1]
```diff
@@ -7,6 +7,12 @@
   font-weight: normal;
 }
 
+.null-text {
+  color: $primary-green;
```
Comment (vedant-jain03) [inline app/assets/stylesheets/groups.scss:11]:
```suggestion
  color: $iframe-container-grey;
```
Try this color not primary for the message!

File: config/locales/views/groups/en.yml [hunk 2]
```diff
@@ -22,11 +22,14 @@ en:
       edit_name_btn: "Edit Name"
       primary_mentor_heading: "Primary Mentor:"
       mentor_name_heading: "MENTORS:"
-      group_members_heading: "MEMBERS"
+      group_members_heading: "MEMBERS:"
       add_members_btn: "+ Add Members"
       add_mentors_btn: "+ Add Mentors"
+      no_mentors: "There are no mentors in this group, click + Add Member button to add members."
+      no_members: "There are no members in this group, click + Add Member button to add members."
+      no_assignments: "There are no assignments in the group, click + Add Assignments button to start an assignment."
```
Comment (vedant-jain03) [inline config/locales/views/groups/en.yml:30]:
```suggestion
      no_mentors_html: "There are no mentors in this group, click <strong>+ Add Mentors</strong> button to add mentors."
      no_members_html: "There are no members in this group, click <strong>+ Add Members</strong> button to add members."
      no_assignments_html: "There are no assignments in the group, click <strong>+ Add Assignments</strong> button to start an assignment."
```

Try this one! also do this for `hi.yml`

File: app/views/groups/show.html.erb [hunk 3]
```diff
@@ -52,6 +63,11 @@
       <%= link_to t("groups.show.assignment.add_assignment_btn"), new_group_assignment_path(@group), class: "btn primary-button groups-primary-button" %>
     <% end %>
   </div>
+    <% if @group.assignments.empty? %>
+      <ul>
+        <li class="null-text"><%= t("groups.show.no_assignments") %></li>
+      </ul>
```
Comment (vedant-jain03) [inline app/views/groups/show.html.erb:69]:
```suggestion
  <hr>
  <% if @group.assignments.empty? %>
    <span class="null-text"><%= t("groups.show.no_assignments_html") %></span>
  <% end %>
```

**Comment by VaibhavUpreti** [comment]
Screenshot 2022-12-08 at 2 16 41 PM
This is how it looks

File: app/views/groups/show.html.erb [hunk 4]
```diff
@@ -52,14 +58,15 @@
...
   <table class="table table-borderless groups-table">
     <thead class="groups-table-heading">
       <tr>
-        <th><%= t("groups.show.assignment.assignment_table_head.name") %></th>
-        <th><%= t("groups.show.assignment.assignment_table_head.deadline") %></th>
-        <th><%= t("groups.show.assignment.assignment_table_head.grading") %></th>
-        <th><%= t("groups.show.assignment.assignment_table_head.actions") %></th>
         <% unless policy(@group).admin_access? %>
           <th><%= t("groups.show.assignment.assignment_table_head.grade") %></th>
         <% end %>
```
Comment (vedant-jain03) [inline app/views/groups/show.html.erb:72]: Why remove this?
```
  <hr>
  <table class="table table-borderless groups-table">
    <% if @group.assignments.empty? %>
      <span class="null-text"><%= t("groups.show.no_assignments_html") %></span>
    <% else %>
      <thead class="groups-table-heading">
        <tr>
          <th><%= t("groups.show.assignment.assignment_table_head.name") %></th>
          <th><%= t("groups.show.assignment.assignment_table_head.deadline") %></th>
          <th><%= t("groups.show.assignment.assignment_table_head.grading") %></th>
          <th><%= t("groups.show.assignment.assignment_table_head.actions") %></th>
          <% unless policy(@group).admin_access? %>
            <th><%= t("groups.show.assignment.assignment_table_head.grade") %></th>
          <% end %>
        </tr>
      </thead>
    <% end %>
```
Try this.
What we want is if there are no assignment render no_assignment_message else render table head and body!

File: config/locales/views/groups/hi.yml [hunk 5]
```diff
@@ -25,6 +25,9 @@ hi:
       group_members_heading: "सदस्य"
       add_members_btn: "+ सदस्य ऐड करे"
       add_mentors_btn: "+ मेंटर ऐड करे"
+      no_mentors: "इस समूह में कोई संरक्षक नहीं हैं, सदस्यों को जोड़ने के लिए  <strong>+ सदस्य जोड़ें</strong> बटन पर क्लिक करें।"
+      no_members: "समूह में कोई सदस्य नहीं है, सदस्यों को जोड़ने के लिए  <strong>+ पथप्रदर्शक जोड़ें</strong> बटन पर क्लिक करें।"
+      no_assignments: "समूह में कोई असाइनमेंट नहीं है, असाइनमेंट शुरू करने के लिए <strong>+ असाइनमेंट जोड़ें</strong> बटन पर क्लिक करें।"
```
Comment (vedant-jain03) [inline config/locales/views/groups/hi.yml:30]: Key naming in `en.yml` is not the same here!

**Review by vedant-jain03 — CHANGES_REQUESTED** [review]

File: app/views/groups/show.html.erb (hunk 4 above)
Comment (VaibhavUpreti) [reply app/views/groups/show.html.erb:72]: Oh sorry I misread it and acted in a hurry. Correcting it 👍

Comment (VaibhavUpreti) [reply app/views/groups/show.html.erb:72]: Screenshot 2022-12-08 at 7 02 25 PM
Looks like this now

File: app/views/groups/show.html.erb [hunk 6]
```diff
@@ -52,19 +58,28 @@
       <%= link_to t("groups.show.assignment.add_assignment_btn"), new_group_assignment_path(@group), class: "btn primary-button groups-primary-button" %>
     <% end %>
   </div>
-  <hr>
+    <% if @group.assignments.empty? %>
+      <hr>
+      <% if @group.assignments.empty? %>
+        <span class="null-text"><%= t("groups.show.no_assignments_html") %></span>
+      <% end %>
+    <% end %>
```
Comment (vedant-jain03) [inline app/views/groups/show.html.erb:66]: What is the meaning of adding this again?
This renders the warning twice!

File: config/locales/views/groups/en.yml [hunk 7]
```diff
@@ -22,11 +22,14 @@ en:
       edit_name_btn: "Edit Name"
       primary_mentor_heading: "Primary Mentor:"
       mentor_name_heading: "MENTORS:"
-      group_members_heading: "MEMBERS"
+      group_members_heading: "MEMBERS:"
       add_members_btn: "+ Add Members"
       add_mentors_btn: "+ Add Mentors"
+      no_mentors_html: "There are no mentors in this group, click <strong>+ Add Mentors</strong> button to add mentors."
+      no_members_html: "There are no members in this group, click <strong>+ Add Members</strong> button to add members."
+      no_assignments_html: "There are no assignments in the group, click <strong>+ Add Assignments</strong> button to start an assignment."
```
Comment (vedant-jain03) [inline config/locales/views/groups/en.yml:30]:
```suggestion
      no_assignments_html: "There are no assignments in the group, click <strong>+ Add Assignment</strong> button to start an assignment."
```

**Review by vedant-jain03 — CHANGES_REQUESTED** [review]
@VaibhavUpreti

File: app/views/groups/show.html.erb (hunk 6 above)
Comment (VaibhavUpreti) [reply app/views/groups/show.html.erb]: 😅 forgot to test ... blindly committed code review suggestions. Sorry for inconvenience.

File: app/views/groups/show.html.erb [hunk 8]
```diff
@@ -52,19 +58,22 @@
       <%= link_to t("groups.show.assignment.add_assignment_btn"), new_group_assignment_path(@group), class: "btn primary-button groups-primary-button" %>
     <% end %>
   </div>
-  <hr>
```
Comment (vedant-jain03) [inline app/views/groups/show.html.erb:55]: Restore this line!

**Review by vedant-jain03 — CHANGES_REQUESTED** [review]
@VaibhavUpreti LGTM, just a small change!

**Comment by VaibhavUpreti** [comment]
👇
f

**Review by vedant-jain03 — APPROVED** [review]
@VaibhavUpreti Good work 💯 
cc @tachyons

**Review by tachyons — APPROVED** [review]

**Pull request merged**